### PR TITLE
Adding more control on making requests

### LINKED ISSUE #308: Adding more control on sending requests to BotAPI
**Is your feature request related to a problem? Please describe.**
Sometimes it's useful to modify, cancel or delay sending request to BotAPI.

**Describe the solution you'd like**
Adding 2 new middleware triggers, before and after making request.

### PR DESCRIPTION
**Pull request by Forden**
# Description

Fixes #308 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by uwinx** [comment]
@Forden,  Can it be achieved by another approach? Because `Bot` doesn't always run within `Dispatcher`.

**Comment by Forden** [comment]
No, it cannot be achieved any other way, because middlewares are stored in `Dispatcher.middleware.applications`

**Comment by Forden** [comment]
I think the best way (for implementing this feature right now, not in 3.0) is adding if statement to check if bot is connected with dispatcher. If it's not, request will be sent immediately, but if there is a dispatcher - middleware will be triggered.

File: aiogram/bot/base.py [hunk 1]
```diff
@@ -190,8 +191,18 @@ async def request(self, method: base.String,
         :rtype: Union[List, Dict]
         :raise: :obj:`aiogram.exceptions.TelegramApiError`
         """
-        return await api.make_request(self.session, self.__token, method, data, files,
-                                      proxy=self.proxy, proxy_auth=self.proxy_auth, timeout=self.timeout, **kwargs)
+        from aiogram import Dispatcher
+        dp = Dispatcher.get_current()
```
Comment (JrooTJunior) [inline aiogram/bot/base.py:195]: The bot instance should not know anything about the dispatcher. This change breaks everything.

**Pull request closed without merging**